alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * @package Core
+ * @subpackage externalWidgets
+ */
+class embedPlaykitJsAction extends sfAction
+{
+    var $eTagHash = null;
+    var $partner_id = null;
+    var $ui_conf_id = null;
+    var $sourcesPath = null;
+    var $bundle_name = null;
+
+    public function execute()
+    {
+        //Get uiConf ID from QS
+        $this->uiconf_id = $this->getRequestParameter('uiconf_id');
+        if (!$this->uiconf_id)
+            KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, 'uiconf_id');
+
+        // retrieve uiCong Obj
+        $uiConf = uiConfPeer::retrieveByPK($this->uiconf_id);
+        if (!$uiConf)
+            KExternalErrors::dieError(KExternalErrors::UI_CONF_NOT_FOUND);
+
+        //Get partner ID from QS
+        $this->partner_id = $this->getRequestParameter('partner_id', $uiConf->getPartnerId());
+        if (!$this->partner_id)
+            KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, 'partner_id');
+
+        $bundleWebDirPath = kConf::get('playkit_js_bundles_path');
+        $bundleBuilderPath = kConf::get('bundle_builder_cli_path');
+        $this->sourcesPath = kConf::get('playkit_js_sources_path');
+
+        //Get bundle configuration stored in conf_vars
+        $bundleConfig = $uiConf->getConfVars();
+
+        //convert string to json
+        $bundleConfig_json = json_decode($bundleConfig, true);
+
+        //if latest/beta version required set version number in config obj
+        $isLatestVersionRequired = strpos($bundleConfig, "{latest}");
+        $isBetaVersionRequired = strpos($bundleConfig, "{beta}");
+        if ($isLatestVersionRequired || $isBetaVersionRequired) {
+            $bundleConfig_json = $this->setLatestOrBetaVersionNumber($bundleConfig_json);
+        }
+
+        //sort bundle config by key
+        ksort($bundleConfig_json);
+
+        //create base64 bundle name from json config
+        $config_str = json_encode($bundleConfig_json);
+        $this->bundle_name = base64_encode($config_str);
+
+        //check if bundle already exists
+        $bundle_path = $bundleWebDirPath . $this->bundle_name . ".min.js";
+        if (file_exists($bundle_path)) {
+            $bundleContent = $this->getbundleContent($bundle_path);
```

Comment:
$bundleContent is created anyway from $this->getbundleContent($bundle_path), so it can be taken of of the if clause.